src/outputErlangCode.ml

```diff
@@ -167,9 +167,9 @@ let stringify_base_constant (bc : base_constant) =
       else
         assert false
 
-  | BinaryByString(s) -> Printf.sprintf "<<\"%s\">>" (String.escaped s)
+  | BinaryByString(s) -> Printf.sprintf "<<\"%s\"/utf8>>" s
   | BinaryByInts(ns)  -> Printf.sprintf "<<%s>>" (ns |> List.map string_of_int |> String.concat ", ")
-  | String(s)         -> Printf.sprintf "\"%s\"" (String.escaped s)
```

Comment:
I understand that this PR changes what `s` is in `BinaryByString(s)` and `String(s)`; it has been a genuine byte sequence, but after the change it will be a byte sequence where double-quote symbols and backslashes have already been escaped.

This policy does not seem, at least at first glance, so acceptable IMHO, because it leads to a tight coupling between the lexer and the code generator. Do you have some strong opinion to make this change?

Reply:
I agree there is coupling introduced, and relying on the Erlang compiler more. I think an alternative would be to design sesterl's own escaping sequences rules, implement parsing them accordingly, and implement a proper generator for Erlang's escape sequences (because what OCaml's String.escaped was doing was not correct). I didn't want to commit to a design before you give it a thought as to how you would like escape sequences to look like (the same or different than erlang's).

Reply:
> (because what OCaml's String.escaped was doing was not correct)

Ah, I understand that `String.escaped` did not work as intended for characters outside the US-ASCII printable range:

```
utop # String.escaped "あ" |> String.to_seq |> List.of_seq;;
- : char list = ['\\'; '2'; '2'; '7'; '\\'; '1'; '2'; '9'; '\\'; '1'; '3'; '0']
```

> I didn't want to commit to a design before you give it a thought as to how you would like escape sequences to look like (the same or different than erlang's).

I see. Thank you for your kind consideration. Currently, I think of starting with the following relatively minimal rules (although we should probably develop the spec further especially to handle non-printable characters with ease):

* Escape sequences common to binary literals and string ones:
  - `\n`: newline
  - `\r`: carriage return
  - `\t`: tab
  - `\\`: backslash
  - `\"`: double quote
  - `\'`: single quote
* Allowed only in string literals:
  - `"` (without `\` for escape)
* Allowed only in binary literals:
  - `'` (without `\` for escape)

Reply:
Should there be a parser error when someone tries to use an unsupported escape sequence?

Reply:
Yes, it seems safer to do so IMHO.

Reply:
I made some progress but I don't know how the correct escape_string function should look like, I don't have much experience working with strings / uchars in OCaml - I would appreciate any hints!

Reply:
This should be ready for another look!